### DISCUSSION TITLE
api_server_authorized_ip_ranges in Teraform deprecated and causing false positives

### DISCUSSION THREAD

**Discussion by huornlmj** [aquasecurity/trivy#4331] [Bugs]
## Description
Trivy will ignore "authorized_ip_ranges" and will complain if it doesn't find the now deprecated term "api_server_authorized_ip_ranges"

```
CRITICAL: Cluster does not limit API access to specific IP addresses.
```

## What did you expect to happen?
Not get a false positive, and for Trivy to observe the newer construct as a pass

## What happened instead?
A failed scan

**Comment by itaysk** [comment]
just to clarify, you are referring to check `AVD-AZU-0041` which asserts the following argument: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#authorized_ip_ranges
correct?

**Comment by huornlmj** [comment]
Yes @itaysk that's correct.

**Comment by huornlmj** [comment]
This is very similar to https://github.com/aquasecurity/defsec/issues/1142 in defsec

**Comment by simar7** [comment]
hi @huornlmj - I've cut https://github.com/aquasecurity/trivy/issues/4624 to address this issue.
